base_location/models/res_partner.py

```diff
@@ -34,6 +34,17 @@ def _onchange_country_id(self):
         res = super()._onchange_country_id()
         if self.zip_id and self.zip_id.city_id.country_id != self.country_id:
             self.zip_id = False
+        if self.country_id:
+            city_zip_domain = {
+                "zip_id": [("city_id.country_id", "=", self.country_id.id)]
+            }
+            if isinstance(res, dict):
+                if "domain" in res and isinstance(res["domain"], dict):
+                    res["domain"].update(city_zip_domain)
+                else:
+                    res["domain"] = city_zip_domain
```

Comment:
Is this part really needed?

Reply:
`_onchange_country_id` return None in `base` module but it may be inherited in any other module to return a dict that contain or not a domain key.

At this point I can't know in which case we are:
*  res == None
*  res == {'value': {}}
*  res == {'value': {}, 'domain': {}}

Reply:
Yeah, but this case you are covering is:
```python
res == {'value': {}, 'domain': None}
```
which I think it's never going to happen.

Reply:
yes, but essentially  `res == {'value': {}}` (`else "domain" in res`).

Maybe I shouldn't add `isinstance(res["domain"], dict)` but i thought it's safer

Reply:
That's why I insist in simplify code and avoid coverage drop for non existent cases. See my other suggestion.